WORKSPACE

```diff
@@ -17,3 +17,15 @@ load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_depe
 go_rules_dependencies()
 
 go_register_toolchains(go_version = GO_VERSION)
+
+load("@com_google_googleapis//:repository_rules.bzl", "switched_rules_by_language")
```

Comment:
You should be able to roll this into `api_dependencies` in `bazel/repositories.bzl` (not `api/bazel/repositories.bzl`).

Reply:
`api_dependencies` has `http_archive` for com_google_googleapis, so that won't work. We can't load from a repo before it's http_archive.

Reply:
and the naming is confusing. bazel/api_repositories.bzl does not actually load API repositories, so I get why one might get confused (I did).

Reply:
Hey Lizan, do you still think it's possible to add a load from googleapis without WORKSPACE change? I don't see a way, since googleapis is loaded as one of the last calls in there.

Reply:
Hmm I see what you're saying, I think we want to somehow roll this into a `bzl` file, @htuch do you have any idea? Just roll additional dependencies (those WORKSPACE rules after `envoy_dependencies`) in an additional `bzl` file should work but still somewhat ugly.

Reply:
This is actually an interesting problem with bazel. All these steps have to be linearized into separate invocations from WORKSPACE:
- the hack to create @envoy_api
- load stuff from @envoy_api
- load stuff from the stuff loaded from @envoy_api

So we need at least 3 separate bazel functions, and we have two only right now.

Reply:
Gentle ping?

Reply:
Yeah I think that would work, do you mind take a stab?

Reply:
I don't understand the envoy_api hack well, I'd hope @htuch can help there. Either way, this is going to cause churn in importing repositories, even if we change from 2 calls to 3 calls.

Reply:
Yes but less churning in in the future :)

Reply:
Done, PTAL